src/Compiler/Microsoft.AspNetCore.Razor.Language/test/TestFiles/IntegrationTests/ComponentDesignTimeCodeGenerationTest/AtAtHandled/TestComponent.ir.txt

```diff
@@ -0,0 +1,20 @@
+﻿Document - 
+    NamespaceDeclaration -  - Test
+        UsingDirective - (3:1,1 [20] ) - global::System
+        UsingDirective - (26:2,1 [40] ) - global::System.Collections.Generic
+        UsingDirective - (69:3,1 [25] ) - global::System.Linq
+        UsingDirective - (97:4,1 [36] ) - global::System.Threading.Tasks
+        UsingDirective - (136:5,1 [45] ) - global::Microsoft.AspNetCore.Components
+        ClassDeclaration -  - public partial - TestComponent - global::Microsoft.AspNetCore.Components.ComponentBase - 
+            DesignTimeDirective - 
+            CSharpCode - 
+                IntermediateToken -  - CSharp - #pragma warning disable 0414
+            CSharpCode - 
+                IntermediateToken -  - CSharp - private static object __o = null;
+            CSharpCode - 
+                IntermediateToken -  - CSharp - #pragma warning restore 0414
+            MethodDeclaration -  - protected override - void - BuildRenderTree
+                CSharpCode - (2:0,2 [67] x:\dir\subdir\Test\TestComponent.cshtml)
+                    LazyIntermediateToken - (2:0,2 [67] x:\dir\subdir\Test\TestComponent.cshtml) - CSharp -  var validationMessage = @Html.ValidationMessage("test", "", new { 
+                CSharpCode - (70:0,70 [39] x:\dir\subdir\Test\TestComponent.cshtml)
+                    LazyIntermediateToken - (70:0,70 [39] x:\dir\subdir\Test\TestComponent.cshtml) - CSharp - @class = "invalid-feedback" }, "div"); 
```

Comment:
This is the crux of the change. It's best shown by this diff: https://github.com/dotnet/razor/pull/10752/commits/8663ad3066d025117755a1ed74aa04d3e2f20253#diff-a89dc29d4e0f7df972485727da00b99a98e3fe77929e68f75d1bc6cb71f621cc. We go from one statement to 2 statements, with an ephemeral node splitting them.

Reply:
Yes, I see. I'm just wondering whether that's needed, or it would be possible and better to have a single statement rather than two. E.g., aren't there going to be problems with the statements split into two like that (e.g., what if some diagnostic spans both of the generated statements)? But it's just a question / nit considering the change is a backwards-compat thing only.

Reply:
This is essentially replicating what the old code would have here; I'd rather keep it as is, since this is a backcompat issue.